app/src/marketplace/components/StreamPreviewPage/StreamLivePreview/index.jsx

```diff
@@ -30,160 +33,67 @@ export type DataPoint = {
 
 type Props = {
     streamId: ?StreamId,
-    authApiKeyId: ?ResourceKeyId,
     selectedDataPoint: ?DataPoint,
     onSelectDataPoint: (DataPoint, ?boolean) => void,
     run?: boolean,
     userpagesPreview?: boolean,
     hasData?: () => void,
 }
 
-type State = {
-    visibleData: Array<DataPoint>,
-    subscriptionError: boolean,
-    dataError: boolean,
-    mobileTableColumnIndex: number,
-}
-
-let cachedClient: ?StreamrClient
-
 const LOCAL_DATA_LIST_LENGTH = 10
 
-export class StreamLivePreview extends Component<Props, State> {
-    static defaultProps = {
-        run: true,
-    }
-
-    state = {
-        visibleData: [],
-        subscriptionError: false,
-        dataError: false,
-        mobileTableColumnIndex: 0,
-    }
-
-    componentDidMount() {
-        this.createClientAndSubscribe()
-    }
-
-    componentWillUnmount = () => {
-        this.unsubscribe()
-        this.updateDataToState.cancel()
-    }
-
-    onData = (dataPoint: DataPoint) => {
-        if (this.props.hasData && this.state.visibleData.length === 0) {
-            this.props.hasData()
+const prettyPrintData = (data: ?{}, compact: boolean = false) => stringifyObject(data, {
+    indent: '  ',
+    inlineCharacterLimit: compact ? Infinity : 5,
+})
+
+const StreamLivePreview = ({
+    streamId,
+    selectedDataPoint,
+    onSelectDataPoint,
+    userpagesPreview,
+    hasData,
+    run = true,
+}: Props) => {
+    const dataRef = useRef([])
+    const [visibleData, setVisibleData] = useState([])
+    const [dataError, setDataError] = useState(false)
+    const [mobileTableColumnIndex, setMobileTableColumnIndex] = useState(0)
+    const { hasLoaded, client } = useContext(ClientContext)
+
+    const updateDataToState = useCallback(throttle((data) => {
+        if (hasData && visibleData.length === 0) {
+            hasData()
         }
+        setVisibleData([...data])
 
-        if (this.props.run) {
-            this.data.unshift(dataPoint)
-            this.data.length = Math.min(this.data.length, LOCAL_DATA_LIST_LENGTH)
-            this.updateDataToState(this.data)
+        if (!selectedDataPoint && data.length) {
+            onSelectDataPoint(data[0], true)
         }
-    }
-
-    onMobileTableColumnIndexChange = (index: number) => {
-        this.setState({
-            mobileTableColumnIndex: index,
-        })
-    }
-
-    dataColumn: ?HTMLTableCellElement = null
+    }, 100), [hasData, selectedDataPoint])
```

Comment:
Curious about this throttle, probably not a big deal but I wonder if it could be too slow for a tram like stream.

Reply:
It was there before so I left it - but what get's throttled is the display of the value not the data handler. For a preview it probably doesn't matter if not all the messages are captured.